dadi/web
# Feature/redis errors

#75 

Currently, this:
+ Correctly handles redis disconnects, using the newest API (retry_strategy)
+ Falls back to directory based caching
+ Handles timeouts (using a multiplicative timeout strategy up to 4s)
+ Handles cases where redis repeatedly connects and disconnects
+ Has a ton of logging

To test, my setup has been:
+ Spin up redis in docker or a vm of your choice
+ Spin up a web/api instance
+ Do a hard shutdown of redis at the vm/container level, not the service
+ Reboot or observe as needed

Open issues:
+ Handling repeated disconnects/reconnects over a long period of time.

Currently we just track total connects/disconnects, and if we go over a threshold (currently set to three), we scrub it and go back to directory based caching. However, if the connects/disconnects happen over a long period of time, then that probably doesn't indicate serious issues. How do we want to handle this logic?

+ General timings/thresholds

I've gotten my base values from the ruby redis implementation (they use a timeout of 5 seconds total). Do we want to be more aggressive in our timeout potentially?
